### PR TITLE
Feature/mds 1091 frontend for user managment

### PR DESCRIPTION
**Pull request by Jsyro**
Add and delete minespace users. 

hard deletes and no edit for simplicity. Aaron's eyes only when deployed to prod.

### CODE REVIEW

File: frontend/src/components/admin/MinespaceUserList.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+const propTypes = {
+  minespaceUsers: PropTypes.arrayOf(CustomPropTypes.minespaceUser),
+  mines: PropTypes.arrayOf(CustomPropTypes.mineName),
+  handleDelete: PropTypes.func,
+};
+
+const defaultProps = {
+  minespaceUsers: [],
+  mines: [],
+  handleDelete: null,
```
Comment (mmumby) [inline frontend/src/components/admin/MinespaceUserList.js]: null is not a valid default prop for a function, 
use: 

```
handleDelete:  ()=> {};
```

File: frontend/src/components/admin/MinespaceUserManagement.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  handleDelete = (userId) => {
+    this.props.deleteMinespaceUser(userId).then(() => {
+      this.props.fetchMinespaceUsers();
+    });
+  };
+
+  render() {
+    // console.log(this.props.minespaceUsers);
+    // console.log(this.props.mines);
```
Comment (mmumby) [inline frontend/src/components/admin/MinespaceUserManagement.js]: remove

File: frontend/src/components/admin/MinespaceUserList.js (hunk 1 above)
Comment (Jsyro) [reply frontend/src/components/admin/MinespaceUserList.js]: thanks, fixed.

File: frontend/src/components/admin/MinespaceUserManagement.js (hunk 2 above)
Comment (Jsyro) [reply frontend/src/components/admin/MinespaceUserManagement.js]: good catch, thanks.

File: frontend/src/customPropTypes/mines.js [hunk 3]
```diff
@@ -26,3 +26,11 @@ export const minePageData = shape({
   total: PropTypes.numnber,
   total_pages: PropTypes.numnber,
 });
+
+export const mineName = shape({
```
Comment (mmumby) [inline frontend/src/customPropTypes/mines.js]: `mineName` is very specific for an object that contains more than a name. This should be renamed.

File: frontend/src/components/admin/NewMinespaceUser.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    });
+  };
+
+  render() {
+    return (
+      <div>
+        <h3>Add BCEID User</h3>
+        {this.props.mines.mines && (
+          <AddMinespaceUser
+            mines={this.props.mines.mines.map((mine) => ({
```
Comment (mmumby) [inline frontend/src/components/admin/NewMinespaceUser.js]: The data should be transformed before passing it in as a prop.

File: frontend/src/components/admin/MinespaceUserManagement.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  mines: PropTypes.arrayOf(CustomPropTypes.mineName),
+  minespaceUsers: PropTypes.arrayOf(CustomPropTypes.minespaceUser),
+  fetchMineNameList: PropTypes.func.isRequired,
+  fetchMinespaceUsers: PropTypes.func.isRequired,
+  deleteMinespaceUser: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  mines: [],
+  minespaceUsers: {},
```
Comment (mmumby) [inline frontend/src/components/admin/MinespaceUserManagement.js]: should this be? `minespaceUsers: []`

File: frontend/src/components/admin/NewMinespaceUser.js (hunk 4 above)
Comment (Jsyro) [reply frontend/src/components/admin/NewMinespaceUser.js]: removed second .mines. i think that's what you were looking for.

File: frontend/src/components/admin/MinespaceUserManagement.js (hunk 5 above)
Comment (Jsyro) [reply frontend/src/components/admin/MinespaceUserManagement.js]: yes it should.

File: frontend/src/customPropTypes/mines.js (hunk 3 above)
Comment (Jsyro) [reply frontend/src/customPropTypes/mines.js]: `mineNameList: getMineNames(state).mines,` existing places that used that object structure use a selector called getMineNames, so i figure the object type should be MineName. I don't disagree and I'd be happy to rename it. But then the selector name and the customproptype don't match in name, which may be awkward.

Comment (mmumby) [reply frontend/src/customPropTypes/mines.js]: ah, okay, I'm following. 

This is fine then.

**Pull request merged**